assets/tests/components/groupedAutocomplete.test.tsx

```diff
@@ -893,4 +894,107 @@ describe("<GroupedAutocompleteFromSearchTextResults/>", () => {
+      searchText
+    ) => {
+      if (searchText === searchText) {
+        return {
+          vehicle: [vehicle],
+          operator: [],
+          run: [],
+        }
+      }
+      return {
+        operator: [],
+        run: [],
+        vehicle: [],
+      }
+    }) as typeof useAutocompleteResults)
+
+    render(
+      <GroupedAutocompleteFromSearchTextResults
+        controlName="Search Suggestions"
+        fallbackOption={autocompleteOption(null)}
+        onSelectVehicleOption={onSelectVehicleOption}
+        searchText={searchText}
+        searchFilters={searchFiltersFactory.build()}
+        onSelectedLocationId={() => {}}
+        onSelectedLocationText={() => {}}
+      />
+    )
+
+    await userEvent.click(option(vehicle.label!).get())
+
+    expect(onSelectVehicleOption).toHaveBeenCalledWith(vehicle)
+  })
+
+  test("when a location autocomplete option is clicked, should fire event 'onSelectedLocationId'", async () => {
+    const onSelectLocationId = jest.fn()
+    const searchText = "123 Test St"
+    const location = locationSearchResultFactory.build({
+      name: "Search Suggestion",
+    })
+    const locationSuggestion = locationSearchSuggestionFactory.build({
+      text: location.name!,
+      placeId: location.id,
+    })
+
+    jest
+      .mocked(useLocationSearchSuggestions)
+      .mockReturnValue([locationSuggestion])
+
+    render(
+      <GroupedAutocompleteFromSearchTextResults
+        controlName="Search Suggestions"
+        fallbackOption={autocompleteOption(null)}
+        onSelectVehicleOption={() => {}}
+        searchText={searchText}
+        searchFilters={searchFiltersFactory.build()}
+        onSelectedLocationId={onSelectLocationId}
+        onSelectedLocationText={() => {}}
+      />
+    )
+
+    await userEvent.click(option(location.name!).get())
+
+    expect(onSelectLocationId).toHaveBeenCalledWith(location.id)
+  })
+
+  test("when a text-only location autocomplete option is clicked, should fire event 'onSelectedLocationText' and close autocomplete", async () => {
+    const onSelectLocationText = jest.fn()
+    const searchText = "123 Test St"
+    const locationSuggestion = locationSearchSuggestionFactory.build({
+      text: "Suggested Search Term",
+      placeId: null,
+    })
+
+    jest
+      .mocked(useLocationSearchSuggestions)
+      .mockReturnValue([locationSuggestion])
+
+    render(
+      <GroupedAutocompleteFromSearchTextResults
+        controlName="Search Suggestions"
+        fallbackOption={autocompleteOption(null)}
+        onSelectVehicleOption={() => {}}
+        searchText={searchText}
+        searchFilters={searchFiltersFactory.build()}
+        onSelectedLocationId={() => {}}
+        onSelectedLocationText={onSelectLocationText}
+      />
+    )
+
+    await userEvent.click(option(locationSuggestion.text).get())
+
+    expect(onSelectLocationText).toHaveBeenCalledWith(locationSuggestion.text)
+
+    expect(listbox().get()).not.toBeVisible()
+  })
```

Comment:
note: I believe this test is failing because this test is testing something that's not the responsibility of the autocomplete component itself. The autocomplete is not responsible for handling it's visibility, which is why this test lived in `searchForm` and not in `groupedAutocomplete`. So I believe tests will pass if you do this/leave this test on the search form instead.

Reply:
Ahhhhaaaaaa. I moved the dropdown openness check to the searchForm, and kept the `onSelectLocationText` check in groupedAutocomplete